File: src/Chapter-6-C99/Questions/Question9.c
//Write a program that calculates the remaining balance on a loan
//after the first, second, and third monthly payments
#include <stdio.h>

int main(void)
{
    float amount, interest, payment;
    float first, second, third;
    int num;
    printf("Enter amount of loan: ");
    scanf("%f", &amount);
    printf("Enter interest rate: ");
    scanf("%f", &interest);
    printf("Enter monthly payment: ");
    scanf("%f", &payment);
    //Convert interest rate to monthly interest rate
    interest = 1+((interest / 100)/12);
    //ask user for the number of payments
    printf("Enter the number of payments: ");
    scanf("%d", &num);
    for (int i=0; i<num; i++)
    {
        amount = (amount*interest)-payment;
    }
    printf("Balance remaining: $%f\n", amount);

    return 0;
}

File: src/Chapter-6-Rust/Questions/Question9.rs
//Write a program that calculates the remaining balance on a loan
//after the first, second, and third monthly payments
#[macro_use] extern crate text_io;
use std::process::Command;
//Need this to use the print!
use std::io::{self, Write};

fn main()
{
	let (mut amount, mut interest, payment, num):(f32, f32, f32, i32);
	print!("Enter amount of loan: ");
	io::stdout().flush().unwrap();
	scan!("{}", amount);
	print!("Enter interest rate: ");
	io::stdout().flush().unwrap();
	scan!("\n{}", interest);
	print!("Enter monthly payment: ");
	io::stdout().flush().unwrap();
	scan!("\n{}", payment);
	//Convert interest rate to monthly interest rate
	interest = 1.0+((interest/100.00)/12.00);
	//ask user for the number of payments
	print!("Enter the number of payments: ");
	io::stdout().flush().unwrap();
	scan!("\n{}", num);
	for _ in 0..num
	{
		amount = (amount*interest)-payment;
	}
	print!("Balance remaining: ${}\n", amount);	
	io::stdout().flush().unwrap();
	let _ = Command::new("cmd.exe").arg("/c").arg("pause").status();
}
